urlconnection/src/main/java/zipkin/reporter/urlconnection/URLConnectionSender.java

```diff
@@ -162,24 +162,42 @@ void send(byte[] body, String mediaType) throws IOException {
     if (compressionEnabled()) {
       connection.addRequestProperty("Content-Encoding", "gzip");
       ByteArrayOutputStream gzipped = new ByteArrayOutputStream();
-      try (GZIPOutputStream compressor = new GZIPOutputStream(gzipped)) {
+      GZIPOutputStream compressor = new GZIPOutputStream(gzipped);
+      try {
         compressor.write(body);
+      } finally {
+        compressor.close();
       }
       body = gzipped.toByteArray();
     }
     connection.setDoOutput(true);
     connection.setFixedLengthStreamingMode(body.length);
     connection.getOutputStream().write(body);
 
-    try (InputStream in = connection.getInputStream()) {
+    skipAllContent(connection);
+  }
+
+  /** This utility is verbose as we have a minimum java version of 6 */
+  static void skipAllContent(HttpURLConnection connection) throws IOException {
+    InputStream in = connection.getInputStream();
+    IOException thrown = skipAndSuppress(in);
+    if (thrown == null) return;
+    InputStream err = connection.getErrorStream();
+    if (err != null) skipAndSuppress(err); // null is possible, if the connection was dropped
+    throw thrown;
+  }
+
+  static IOException skipAndSuppress(InputStream in) {
```

Comment:
had to do this as errorprone is more strict about classes created by retrolambda